apis/pkg/v1/interfaces.go

```diff
@@ -28,6 +28,19 @@ const (
 	// LabelParentPackage is used as key for the owner package label we add to the
 	// revisions. Its corresponding value should be the name of the owner package.
 	LabelParentPackage = "pkg.crossplane.io/package"
+
+	// TODO(negz): Should we propagate the family label up from revision to
+	// provider? It could potentially change over time, for example if the
+	// active revision's label changed for some reason. There's no technical
+	// reason to need it, but being able to list provider.pkg by family seems
+	// convenient.
+
+	// LabelProviderFamily is used as key for the provider family label. This
+	// label is added to any provider that rolls up to a larger 'family', such
+	// as 'family-aws'. It is propagated from provider metadata to provider
+	// revisions, and can be used to select all provider revisions that belong
+	// to a particular family. It is not added to providers, only revisions.
+	LabelProviderFamily = "pkg.crossplane.io/provider-family"
```

Comment:
CC @nullable-eth. I'm going with this as the label (not annotation) that will show up in `crossplane.yaml` for any provider that's part of a family. It will also show up on the revision, per the comment.

Reply:
We're discussion some potential alternatives to family - specifically "cloud" and "vendor".